packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -830,6 +831,13 @@ - (void)startVideoRecordingWithCompletion:(void (^)(FlutterError *_Nullable))com
                                      details:nil]);
       return;
     }
+    // do not call startWriting in didOutputSampleBuffer to prevent state in which
+    // stopVideoRecordingWithCompletion does not send completion when _isRecording is
+    // YES but _videoWriter.status is AVAssetWriterStatusUnknown and video lag at start
```

Comment:
can you also explain that this happens when stop video is called immediately after start, before the first video sample is arrived (if i understand correctly)